File: c#/web/WebApplication1/Controllers/Common/IndexController.cs
using CommonHelper.Helper;
using CommonHelper.Helper.CommonEntity;
using System;
using System.Collections.Generic;
using System.Drawing;
using System.IO;
using System.Web;
using System.Web.Mvc;
using WebApplication1.Entity.Common;
using WebApplication1.Filter.Common;
using WebApplication1.Service;
namespace WebApplication1.Controllers.Common
{
    /// <summary>
    /// 通用控制器
    /// </summary>
    public class IndexController : FastController
    {
        public SystemService SystemService { set; get; }

        public RealTimeInitService RealTimeInitService { set; get; }

        /// <summary>
        /// 上传文件所允许的路径
        /// </summary>
        public static HashSet<string> _allowPath { set; get; }

        /// <summary>
        /// 等待池表
        /// </summary>
        public static HashSet<string> WaitPoolSet { set; get; }

        /// <summary>
        /// 用户和等待池表，确保每个用户只能在同一个等待池上等待。
        /// </summary>
        public static HashSet<string> UsernameAndPoolSet { set; get; }

        static IndexController()
        {
            _allowPath = new HashSet<string>();
            _allowPath.Add("test");

            WaitPoolSet = new HashSet<string>();
            WaitPoolSet.Add("newsAlarm");

            UsernameAndPoolSet = new HashSet<string>();
        }

        /// <summary>
        /// 加载登陆后数据
        /// </summary>
        /// <returns></returns>
        public JsonResult LoadLoginData()
        {
            Dictionary<string, object> data = new Dictionary<string, object>();
            data.Add("leftMenus", SystemService.LoadLeftMenus());
            return MyJson(new Result { code = 0, data = data });
        }

        /// <summary>
        /// 匿名的实时检测最新版本号，如果发现版本号发生变化，则马上
        /// 返回，如果没变化，则等待30秒或30秒内版本号发生变化后返回
        /// （需要传入认证参数：createDate（yyyyMMddHHmmss）、r（随机字符串）、）
        /// </summary>
        /// <returns></returns>
        [AllowAnonymous]
        [Sign(new string[] { "realTimePool", "realTimeVersion", "createDate", "r" })]
        public JsonResult AnonymousRealTime(string realTimePool, string realTimeVersion,string signKey)
        {
            return RealTime(realTimePool, realTimeVersion,/*SystemService.LoadUsernameBySignKey(signKey)*/"zhang");
        }


		private JsonResult RealTime(string realTimePool,string realTimeVersion,string username)
		{
			if (WaitPoolSet.Contains(realTimePool))
            {
                string newestVersion;
                bool initRet = ThreadHelper.CompareControllerVersion(realTimePool, realTimeVersion, out newestVersion);
                if (realTimeVersion == null)
                {
                    initRet = RealTimeInitService.Init(realTimePool, username);
                }
                if (initRet)
                {
                    string usernameAndPoolKey = username + realTimePool;
                    if (UsernameAndPoolSet.Contains(usernameAndPoolKey))
                    {
                        return MyJson(new Result { code = -1, msg = "当前用户已有线程在等待池中，无法向等待池放入新线程。" });
                    }
                    else
                    {
                        lock (UsernameAndPoolSet)
                        {
                            if (UsernameAndPoolSet.Contains(usernameAndPoolKey))
                            {
                                return MyJson(new Result { code = -1, msg = "当前用户已有线程在等待池中，无法向等待池放入新线程。" });
                            }
                            else
                            {
                                UsernameAndPoolSet.Add(usernameAndPoolKey);
                            }
                        }
                    }
                    ThreadHelper.BatchWait(realTimePool, 60000);
                    lock (UsernameAndPoolSet)
                    {
                        UsernameAndPoolSet.Remove(usernameAndPoolKey);
                    }
					initRet = ThreadHelper.CompareControllerVersion(realTimePool, realTimeVersion, out newestVersion);
                    return MyJson(new Result { code = initRet?1:0,data=new Dictionary<string, string>
                        {
                            ["realTimePool"]= realTimePool,
                            ["realTimeVersion"]= newestVersion
                        }
                    });
                }
                else
                {
                    return MyJson(new Result { code = 0 ,data = new Dictionary<string, string>
                        {
                            ["realTimePool"] = realTimePool,
                            ["realTimeVersion"] = newestVersion
                        }
                    });
                }
            }
            else
            {
                return MyJson(new Result { code = -1,msg = "该等待池未开放。" });
            }
		}

        /// <summary>
        /// 实时检测最新版本号，如果发现版本号发生变化，则马上
        /// 返回，如果没变化，则等待30秒或30秒内版本号发生变化后返回
        /// </summary>
        /// <param name="realTimePool">等待池名称</param>
        /// <param name="realTimeVersion">版本号</param>
        /// <returns>返回结果</returns>
        public JsonResult RealTime(string realTimePool,string realTimeVersion)
        {
			return RealTime(realTimePool,realTimeVersion,User.Identity.Name);   
        }

        /// <summary>
        /// 加载最新提醒消息
        /// </summary>
        /// <param name="version"></param>
        /// <returns></returns>
        public JsonResult LoadNewsAlarm()
        {
            return MyJson(new Result { code = 0, data = SystemService.LoadNewsAlarm() });
        }

        /// <summary>
        /// 加载树节点测试方法，正式上线时删除
        /// </summary>
        /// <returns></returns>
        public JsonResult LoadTreeNode()
        {
            List<TreeFormNode> treeNodeList = new List<TreeFormNode>();
            treeNodeList.Add(new TreeFormNode
            {
                Id = "01",
                Name = "广东"
            });
            treeNodeList.Add(new TreeFormNode
            {
                Id = "02",
                Name = "佛山"
            });
            TreeFormNode treeNode;
            treeNodeList.Add(treeNode = new TreeFormNode
            {
                Id = "03",
                Name = "顺德"
            });
            treeNode.Children.Add(new TreeFormNode
            {
                Id = "31",
                Name = "陈村"
            });
            treeNode.Children.Add(new TreeFormNode
            {
                Id = "32",
                Name = "陳村"
            });
            treeNode.Children.Add(new TreeFormNode
            {
                Id = "33",
                Name = "陳邨"
            });
            treeNode.Children.Add(new TreeFormNode
            {
                Id = "34",
                Name = "陈村1"
            });
            treeNode.Children.Add(new TreeFormNode
            {
                Id = "35",
                Name = "陳村1"
            });
            treeNode.Children.Add(new TreeFormNode
            {
                Id = "36",
                Name = "陳邨1"
            });
            treeNode.Children.Add(new TreeFormNode
            {
                Id = "37",
                Name = "陈村2"
            });
            treeNode.Children.Add(new TreeFormNode
            {
                Id = "38",
                Name = "陳村2"
            });
            treeNode.Children.Add(new TreeFormNode
            {
                Id = "39",
                Name = "陳邨2"
            });
            treeNode.Children.Add(new TreeFormNode
            {
                Id = "40",
                Name = "陈村3"
            });
            treeNode.Children.Add(new TreeFormNode
            {
                Id = "41",
                Name = "陳村3"
            });
            treeNode.Children.Add(new TreeFormNode
            {
                Id = "42",
                Name = "陳邨3"
            });
            treeNode.Children.Add(new TreeFormNode
            {
                Id = "43",
                Name = "陈村4"
            });
            treeNode.Children.Add(new TreeFormNode
            {
                Id = "44",
                Name = "陳村4"
            });
            treeNode.Children.Add(new TreeFormNode
            {
                Id = "45",
                Name = "陳邨4"
            });
            treeNode.Children.Add(new TreeFormNode
            {
                Id = "46",
                Name = "陈村5"
            });
            treeNode.Children.Add(new TreeFormNode
            {
                Id = "47",
                Name = "陳村5"
            });
            treeNode.Children.Add(new TreeFormNode
            {
                Id = "48",
                Name = "陳邨5"
            });
            return MyJson(new Result { code = 0, data = treeNodeList });
        }

        /// <summary>
        /// 上传单张图片
        /// </summary>
        /// <returns></returns>
        [SizeFilter(Size = 3145728, Msg = "上传文件大小不能超过3M，可通过压缩减少文件大小。")]
        [OperCount(CountLimit = 10,ClearMillisecond = 60000)]
        public JsonResult UploadSingleImage(HttpPostedFileBase fileUpload, string pathName)
        {
            if (_allowPath.Contains(pathName))
            {
                Image image = null;
                try
                {
                    image = Image.FromStream(fileUpload.InputStream);
                }
                catch
                {
                    return MyJson(new Result { code = -1, msg = "上传的文件可能不是图片文件，无法处理。" });
                }
                try
                {
                    string imgName = FileHelper.SaveImageBySha1(image, $"{Server.MapPath("~/uploadFiles/")}{pathName}");
                    using (Image img = Image.FromFile($"{Server.MapPath("~/uploadFiles/")}{pathName}{Path.DirectorySeparatorChar}{imgName}"))
                    {
                        using (Image thumbnailImg = img.GetThumbnailImage(150, img.Height * 150 / img.Width, () => false, IntPtr.Zero))
                        {
                            string thumbnailName = FileHelper.SaveImageBySha1(thumbnailImg, $"{Server.MapPath("~/uploadFiles/")}{pathName}");
                            return MyJson(new Result { code = 0, data = new { thumbnailName = thumbnailName, imgName = imgName, imgWidth = img.Width, imgHeight = img.Height } });
                        }
                    }
                }
                catch (Exception ex)
                {
                    return MyJson(new Result { code = -1, msg = ex.Message });
                }
            }
            else
            {
                return MyJson(new Result { code = -1, msg = "该路径不允许上传文件。" });
            }
        }

        /// <summary>
        /// 多个文件上传
        /// </summary>
        /// <param name="fileUploads"></param>
        /// <param name="pathName"></param>
        /// <returns></returns>
        [SizeFilter(Size = 52428800, Msg = "上传文件大小不能超过50M，可通过压缩减少文件大小。")]
        [OperCount(CountLimit = 100, ClearMillisecond = 60000)]
        public JsonResult UploadFiles(HttpPostedFileBase fileUploads, string pathName)
        {
            if (_allowPath.Contains(pathName))
            {
                return MyJson(new Result { code = 0, data = FileHelper.SaveFileNameBySha1(fileUploads.InputStream, $"{Server.MapPath("~/uploadFiles/")}{pathName}") });
            }
            else
            {
                return MyJson(new Result { code = -1, msg = "该路径不允许上传文件。" });
            }
        }

        /// <summary>
        /// 切割单张图片
        /// </summary>
        /// <param name="pathName">图片路径名称</param>
        /// <param name="imgName">图片名称</param>
        /// <param name="imgWidth">图片宽度</param>
        /// <param name="imgHeight">图片高度</param>
        /// <param name="x">左上角切点x坐标</param>
        /// <param name="y">左上角切点y坐标</param>
        /// <param name="w">切割宽度</param>
        /// <param name="h">切割高度</param>
        public JsonResult SingleImageCrop(string pathName, string imgName, int imgWidth, int imgHeight, int x, int y, int w, int h)
        {
            string imgPath = $"{Server.MapPath("~/uploadFiles/")}{pathName}{Path.DirectorySeparatorChar}{imgName}";
            if (System.IO.File.Exists(imgPath))
            {
                Bitmap bitmap = null;
                try
                {
                    bitmap = new Bitmap(imgPath);
                }
                catch
                {
                    return MyJson(new Result { code = -1, msg = "该文件可能不是图片文件，切割失败！" });
                }
                try
                {
                    using (bitmap)
                    {
                        using (Image cutImg = ImageHandleHelper.CutPicByRect(bitmap.GetThumbnailImage(imgWidth, imgHeight, () => false, IntPtr.Zero), x, y, w, h))
                        {

                            using (Image thumbnailImg = cutImg.GetThumbnailImage(150, cutImg.Height * 150 / cutImg.Width, () => false, IntPtr.Zero))
                            {
                                return MyJson(new Result
                                {
                                    code = 0,
                                    data = new
                                    {
                                        thumbnailName = FileHelper.SaveImageBySha1(thumbnailImg, $"{Server.MapPath("~/uploadFiles/")}{pathName}"),
                                        imgName = FileHelper.SaveImageBySha1(cutImg, $"{Server.MapPath("~/uploadFiles/")}{pathName}")
                                    }
                                });
                            }
                        }
                    }
                }
                catch (Exception ex)
                {
                    return MyJson(new Result { code = -1, msg = ex.Message });
                }
            }
            else
            {
                return MyJson(new Result { code = -1, msg = "图片不存在，切割失败！" });
            }
        }

		/// <summary>
        /// 允许匿名访问的显示上传的图片
        /// </summary>
        /// <param name="pathName">图片路径名称</param>
        /// <param name="imgName">图片名称</param>
        [AllowAnonymous]
        [Sign(new string[] { "createDate", "r" , "pathName", "imgName" })]
        [OutputCache(Duration = int.MaxValue)]
        public void AnonymousShowImage(string pathName, string imgName)
        {
            ShowImage(pathName, imgName);
        }

        /// <summary>
        /// 显示上传的图片
        /// <param name="pathName">图片路径名称</param>
        /// <param name="imgName">图片名称</param>
        /// </summary>
        [OutputCache(Duration = int.MaxValue)]
        public void ShowImage(string pathName, string imgName)
        {
            Response.ContentType = "image/jpeg";
            string imgPath = $"{Server.MapPath("~/uploadFiles/")}{pathName}{Path.DirectorySeparatorChar}{imgName}";
            if (System.IO.File.Exists(imgPath))
            {
                using (Stream stream = System.IO.File.OpenRead(imgPath))
                {
                    StreamHelper.CopyStream(stream, Response.OutputStream, false);
                }
            }
            else
            {
                Response.StatusCode = 404;
            }
        }

        /// <summary>
        /// 加载地区选择json
        /// </summary>
        /// <returns></returns>
        [OutputCache(Duration = int.MaxValue)]
        public JsonResult AreaSelect()
        {
            return MyJson(new Result { code = 0, data = SystemService.LoadAreaTree() });
        }

        /// <summary>
        /// 下载文件
        /// </summary>
        /// <param name="pathName">文件路径名称</param>
        /// <param name="fileName">文件sha1名称</param>
        /// <param name="fileDesc">文件的中文描述</param>
        /// <returns></returns>
        [OutputCache(Duration = int.MaxValue)]
        public FilePathResult DownFile(string pathName, string fileName, string fileDesc)
        {
            string filePath = $"{Server.MapPath("~/uploadFiles/")}{pathName}{Path.DirectorySeparatorChar}{fileName}";
            if (System.IO.File.Exists(filePath))
            {
                return File(filePath, MimeMapping.GetMimeMapping(filePath), fileDesc);
            }
            else
            {
                Response.StatusCode = 404;
                return null;
            }
        }
    }
}

File: java/web/src/main/java/web/template/controller/common/IndexController.java
package web.template.controller.common;
import java.awt.Image;
import java.awt.Rectangle;
import java.io.File;
import java.io.FileInputStream;
import java.io.FileNotFoundException;
import java.io.IOException;
import java.io.InputStream;
import java.util.ArrayList;
import java.util.HashMap;
import java.util.HashSet;
import java.util.List;
import java.util.Map;
import java.util.Set;
import javax.imageio.ImageIO;
import javax.servlet.http.HttpServletRequest;
import javax.servlet.http.HttpServletResponse;
import org.apache.commons.io.IOUtils;
import org.apache.shiro.SecurityUtils;
import org.apache.shiro.subject.Subject;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.util.FileCopyUtils;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RequestParam;
import org.springframework.web.bind.annotation.RestController;
import org.springframework.web.multipart.MultipartFile;

import com.txj.common.FileHelper;
import com.txj.common.ImageHandleHelper;
import com.txj.common.ThreadHelper;

import web.template.entity.common.Result;
import web.template.entity.common.TreeFormNode;
import web.template.service.common.RealTimeInitService;
import web.template.service.common.SystemService;
@RestController
@RequestMapping("/index")
public class IndexController extends BaseController {
	
	@Autowired
	private SystemService systemService;
	
	@Autowired
	private RealTimeInitService realTimeInitService;
	
	private Set<String> allowPath;
	
	/**
	 * 等待池表
	 */
	public Set<String> WaitPoolSet;
	
	/**
	 * 用户和等待池表，确保每个用户只能在同一个等待池上等待。
	 */
	public HashSet<String> UsernameAndPoolSet;
	
	public IndexController(){
		WaitPoolSet=new HashSet<String>();
		WaitPoolSet.add("newsAlarm");
		
		allowPath=new HashSet<String>();
		allowPath.add("test");
		
		UsernameAndPoolSet = new HashSet<String>();
	}
	
	/**
	 * 
	 * @param realTimePool
	 * @param realTimeVersion
	 * @param username
	 * @return
	 */
	private Result realTime(String realTimePool,String realTimeVersion,String username){
		if(WaitPoolSet.contains(realTimePool)){
			String[] newestVersion=new String[1];
			boolean initRet = ThreadHelper.compareControllerVersion(realTimePool, realTimeVersion, newestVersion);
			if (realTimeVersion == null){
				initRet = realTimeInitService.init(realTimePool,username);
			}
			Map<String,String> data=new HashMap<String, String>();
			if (initRet){
				ThreadHelper.batchWait(realTimePool, 60000);
				initRet = ThreadHelper.compareControllerVersion(realTimePool, realTimeVersion, newestVersion);
				data.put("realTimePool", realTimePool);
				data.put("realTimeVersion", newestVersion[0]);
				return new Result(1,null,data);
			}else{
				data.put("realTimePool", realTimePool);
				data.put("realTimeVersion", newestVersion[0]);
				return new Result(0,null,data);
			}
		}else{
			return new Result(-1,"该等待池未开放。",null);
		}
	}
	
	/**
	 * 
	 * @param realTimePool
	 * @param realTimeVersion
	 * @param signKey
	 * @return
	 */
	public Result anonymousRealTime(String realTimePool, String realTimeVersion, String signKey)
    {
        return realTime(realTimePool, realTimeVersion,/*systemService.LoadUsernameBySignKey(signKey)*/"zhang");
    }
	
	/**
	 * 
	 * @param realTimePool
	 * @param realTimeVersion
	 * @return
	 */
	@RequestMapping("/realTime")
	public Result realTime(String realTimePool,String realTimeVersion){
		return realTime(realTimePool,realTimeVersion,username());
	}
	
	/**
	 * 加载地区选择json
	 * @return
	 */
	public Result areaSelect(){
		return null;
	}
	
	/**
	 * 登陆后加载基础数据
	 * @return
	 */
	@RequestMapping("/loadLoginData")
	public Result loadLoginData(){
		Map<String, Object> data = new HashMap<String, Object>();
        data.put("leftMenus", systemService.loadLeftMenus(username()));
		return new Result(0,null,data);
	}
	
	/**
	 * 加载最新消息提醒
	 * @return
	 */
	@RequestMapping("/loadNewsAlarm")
	public Result loadNewsAlarm()
    {
        return new Result(0,null,systemService.loadNewsAlarm());
    }
	
	
	/**
	 * 登出功能
	 * @return
	 */
	@RequestMapping("/logout")
	public Result logout(){
		Subject subject = SecurityUtils.getSubject();
		subject.logout();
		return new Result(0,null,null);
	}
	
	/**
	 * 获取左侧菜单
	 * @return
	 */
	@RequestMapping("/loadLeftMenus")
	public Result loadLeftMenus(){
		return new Result(0,"",systemService.loadLeftMenus(username()));
	}
	
	/**
	 * 显示图片
	 * @param pathName
	 * @param imgName
	 */
	@RequestMapping("/showImage")
	public Result showImage(String pathName,String imgName,HttpServletRequest request,HttpServletResponse response){
		if(allowPath.contains(pathName)){
			InputStream inputStream;
			try {
				File imgFile=realFile("/WEB-INF/uploadFiles/"+pathName+"/"+imgName);
				if(imgFile.exists()){
					inputStream = new FileInputStream(imgFile);
					response.addHeader("Cache-control","max-age="+Integer.MAX_VALUE);
					response.setContentType("image/jpeg");
					FileCopyUtils.copy(inputStream, response.getOutputStream());					
				}else{
					response.setStatus(404);
				}
			} catch (IOException e) {
				e.printStackTrace();
			}
			return null;
		}else {
			return new Result(-1, "该目录不允许查看", null);
		}
	}
	
	/**
	 * 下载文件
	 * @param pathName
	 * @param fileName
	 * @param fileDesc
	 */
	@RequestMapping("/downFile")
	public void downFile(String pathName, String fileName, String fileDesc,HttpServletResponse response){
		File file=realFile("/WEB-INF/uploadFiles/"+pathName+"/"+fileName);
		if(allowPath.contains(pathName) && file.exists()){
			try (FileInputStream fileInputStream=new FileInputStream(file)){
				response.setContentType("application/octet-stream");
				response.setHeader("Content-disposition", "attachment; filename="+ new String(fileDesc.getBytes("utf-8"), "ISO8859-1"));
				IOUtils.copy(fileInputStream, response.getOutputStream());
			} catch (FileNotFoundException e) {
				e.printStackTrace();
			} catch (IOException e) {
				e.printStackTrace();
			}
		}else{
			response.setStatus(404);
		}
	}
	
	/**
	 * 切割单张图片
	 * @param pathName	图片路径名称
	 * @param imgName	图片名称
	 * @param imgWidth	图片宽度
	 * @param imgHeight	图片高度
	 * @param x	左上角切点x坐标
	 * @param y	左上角切点y坐标
	 * @param w	切割宽度
	 * @param h	切割高度
	 * @return	返回切割结果，thumbnailName（切割图）、imgName（缩略图）、
	 */
	@RequestMapping("/singleImageCrop")
	public Result singleImageCrop(String pathName, String imgName,int imgWidth,int imgHeight,int x,int y,int w,int h,HttpServletRequest request){
		try {
			File file=realFile("/WEB-INF/uploadFiles/"+pathName+"/"+imgName);
			if(file.exists()){
				Image scaleImage=ImageHandleHelper.scale(ImageIO.read(file), imgWidth, imgHeight);
				Image cutImage=ImageHandleHelper.cutPicByRect(scaleImage, new Rectangle(x, y, w, h));
				Image thumbnailImg=ImageHandleHelper.scale(cutImage, 150, cutImage.getHeight(null) * 150 / cutImage.getWidth(null));
				Map<String,String> data=new HashMap<String, String>();
				data.put("imgName", FileHelper.SaveImageBySha1(cutImage, realPath("/WEB-INF/uploadFiles/"+pathName)));
				data.put("thumbnailName", FileHelper.SaveImageBySha1(thumbnailImg, realPath("/WEB-INF/uploadFiles/"+pathName)));
				return new Result(0, null, data);
			}else{
				return new Result(-1,"图片不存在，切割失败！",null);
			}
		} catch (IOException e) {
			e.printStackTrace();
			return new Result(-1,e.getMessage(),null);
		}
	}
	
	
	
	/**
	 * 上次单张图片
	 * @param fileUpload	上传图片
	 * @param request	请求对象
	 * @param pathName	路径名称
	 * @return
	 */
	@RequestMapping("/uploadSingleImage")
	public Result uploadSingleImage(@RequestParam("fileUpload")MultipartFile fileUpload,HttpServletRequest request,String pathName){
		if(allowPath.contains(pathName)){			
			try {
				File target=realFile("/WEB-INF/uploadFiles/"+pathName+"/");
				String sha1=FileHelper.SaveInputStreamBySha1(fileUpload.getInputStream(), target.getAbsolutePath());
				Image image=ImageIO.read(new File(target.getAbsoluteFile()+File.separator+sha1));
				Image scaleImg=ImageHandleHelper.scale(image, 150, image.getHeight(null) * 150 / image.getWidth(null));
				String thumbnailName=FileHelper.SaveImageBySha1(scaleImg, target.getAbsolutePath());
				Map<String, Object> dataMap=new HashMap<>();
				dataMap.put("thumbnailName", thumbnailName);
				dataMap.put("imgName", sha1);
				dataMap.put("imgWidth", image.getWidth(null));
				dataMap.put("imgHeight", image.getHeight(null));
				return new Result(0, null, dataMap);
			} catch (IllegalStateException e) {
				e.printStackTrace();
				throw new RuntimeException(e);
			} catch (IOException e) {
				e.printStackTrace();
				throw new RuntimeException(e);
			}
		}else{
			return new Result(-1, "该目录不允许上传！", null);
		}
	}
	
	/**
	 * 上传多个文件
	 * @param fileUploads	上传的文件
	 * @param pathName	路径名称
	 * @return
	 */
	@RequestMapping("/uploadFiles")
	public Result uploadFiles(@RequestParam("fileUploads")MultipartFile fileUploads, HttpServletRequest request,String pathName){
		if (allowPath.contains(pathName))
        {
			try {
				File target = realFile("/WEB-INF/uploadFiles/"+pathName+"/");
				return new Result(0, null, FileHelper.SaveInputStreamBySha1(fileUploads.getInputStream(), target.getAbsolutePath()));
			} catch (IOException e) {
				e.printStackTrace();
				return new Result(-1,e.getMessage(),null);
			}
		}else{			
			return new Result(-1, "该路径不允许上传文件。", null);
		}
	}
	
	
	@RequestMapping("/loadTreeNode")
	public Result loadTreeNode()
    {
        List<TreeFormNode> treeNodeList = new ArrayList<TreeFormNode>();
        TreeFormNode t01=new TreeFormNode();
        {
        	t01.setId("01");
        	t01.setName("广东"); 
        }
        treeNodeList.add(t01);
        TreeFormNode t02=new TreeFormNode();
        {
        	t02.setId("02");
        	t02.setName("佛山");
        }
        treeNodeList.add(t02);
        TreeFormNode treeNode = new TreeFormNode();
        {
        	treeNode.setId("03");
        	treeNode.setName("顺德");
        }
        treeNodeList.add(treeNode);
        TreeFormNode t31=new TreeFormNode();
        {
        	t31.setId("31");
        	t31.setName("陈村");
        }
        treeNode.getChildren().add(t31);
        TreeFormNode t32=new TreeFormNode();
        {
        	t32.setId("32");
        	t32.setName("陳村");
        }
        treeNode.getChildren().add(t32);
        TreeFormNode t33=new TreeFormNode();
        {
        	t33.setId("33");
        	t33.setName("陳邨");
        }
        treeNode.getChildren().add(t33);
        return new Result (0,null ,treeNodeList );
    }
}